gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/avro/MRCompactorAvroKeyDedupJobRunner.java

```diff
@@ -99,7 +108,9 @@ protected void configureJob(Job job) throws IOException {
 
   private void configureSchema(Job job) throws IOException {
     Schema newestSchema = getNewestSchemaFromSource(job);
-    AvroJob.setInputKeySchema(job, newestSchema);
+    if (useSingleInputSchema) {
+      AvroJob.setInputKeySchema(job, newestSchema);
+    }
     AvroJob.setMapOutputKeySchema(job, this.shouldDeduplicate ? getKeySchema(job, newestSchema) : newestSchema);
```

Comment:
If `MapOutputKeyValueSchema` and `MapOutputValueSchema` is still `newestSchema`, I guess the error will still happen in the mapper (instead of RecordReader)? I think you can use `context.getInputSplit().getSchema()` in the mapper as the mapper output key/value schema.